src/main/java/com/google/sps/daos/DatabaseRoomDao.java

```diff
@@ -0,0 +1,75 @@
+package com.google.sps.daos;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import com.google.api.core.ApiFuture;
+import com.google.cloud.firestore.CollectionReference;
+import com.google.cloud.firestore.DocumentReference;
+import com.google.cloud.firestore.Firestore;
+import com.google.firebase.database.DatabaseReference;
+import com.google.firebase.database.FirebaseDatabase;
+import com.google.cloud.firestore.SetOptions;
+import com.google.cloud.firestore.DocumentSnapshot;
+import com.google.cloud.firestore.WriteResult;
+import com.google.firestore.v1.StructuredQuery.CollectionSelector;
+
+public class DatabaseRoomDao implements RoomDao {
+
+  private DatabaseReference roomsDBReference;
+  private Firestore firestoreDb;
+  private String actualRoom;
+  private String gameId;
```

Comment:
Since `gameId` is part of a `Room`, does the servlet need to store a reference to it?

Reply:
Right.. Another way to ask the same question: what happens to the `gameId` and `actualRoom` properties if more than one room exists where games are being played?

Reply:
Each room has its own game? So if you want to play a game it generates its own room, you can't play several games in one room because it's the 'game instance'. But I till delete it from the dao it's not necessary there.

Reply:
Yes, but this is the implementation of the interface that handles interacting with the database, correct? So this is assuming that only one room/game exists at a time in the whole world, right?